### ISSUE TITLE
Uninformative Exception Message for % 0 exceptions

### ISSUE THREAD

**Issue by pnappa** [souffle-lang/souffle#231]
When Soufflé performs a modulo by 0, it crashes with an uninformative message:

`[1]    99543 floating point exception  souffle -D- err.dl`

This can be reproduced with the following datalog file:

```
.decl crash(in : number) output
crash(x) :- x=5, fc=0, x % fc = 0.
```

It would be important to highlight the source of the error with reference to the original datalog file.

**Comment by lyndonhenry** [comment]
I had a look at this to see if it could be resolved easily, it turns out it cannot. Catching the exception itself is a simple matter of adding an assertion statement in RamExecutor.cpp during visit BinaryOp, however tracing this back to the point in the original datalog program where it occurred is difficult as (insofar as I can tell) that information is lost by the time the operation is actually executed. Moreover this can't be done in the semantic check as, for example in the program above `fc` is not yet instantiated to zero, and so in the `x % fc = 0` expression we do not know that we are performing a modulo by zero. As any solution must involve a check at evaluation time, the only way I can think of doing this is to retain information about the call site of each operator in the datalog program over the entire course of its evaluation -- hopefully there's a better way, as this adds a relatively large overhead. 

In the meantime, I've added said assertion statement to the RamExecutor, and added a new test in evaluation called "divide_by_zero" which has been commented out with a reference to this issue in evaluation.at. This is included in pull request #294.

**Comment by b-scholz** [comment]
I have added a signal handler and a new RAM statement RamDebugInfo that provides the signal handler context.
